src/ln/peer_handler.rs

```diff
@@ -516,10 +521,30 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 														self.initial_syncs_sent.fetch_add(1, Ordering::AcqRel);
 														local_features.set_initial_routing_sync();
 													}
+													//we need to let them know about us, before we can send stuff to them
 													encode_and_send_msg!(msgs::Init {
 														global_features: msgs::GlobalFeatures::new(),
 														local_features,
 													}, 16);
+													if do_they_require_sync {
+														peer.last_synced_channel = 0;
+														let all_messages_tuple = self.message_handler.route_handler.get_next_announcements(peer.last_synced_channel,5);
+														while (all_messages_tuple.0).len()>0 {
+															for i in 0..all_messages_tuple.0.len(){
+																encode_and_send_msg!(all_messages_tuple.0[i].0.clone(), 256);
+																encode_and_send_msg!(all_messages_tuple.0[i].2.clone(), 257);
+																encode_and_send_msg!(all_messages_tuple.0[i].3.clone(), 257);
+																encode_and_send_msg!(all_messages_tuple.0[i].1.clone(), 258);
+															}
+															//we wait to see if the buffer clears before we retrieve more message to send
+															//we push up to 15 message out, 5 remaining means they have cleared 2/3rds
+															while peer.pending_outbound_buffer.len()>5{
+																thread::sleep(time::Duration::from_millis(50));
```

Comment:
We shouldn't sleep here, we should instead mark where we are in the peer and add more messages to the outbound queue when we get a write_event() from the user.